hnn_core/tests/test_feed.py

```diff
@@ -32,41 +29,20 @@ def test_extfeed():
                        params=p_unique[feed_type],
                        gid=0)
         print(feed)  # test repr
+
     # XXX but 'common' (rhythmic) feeds are not
     for ii in range(len(p_common)):  # len == 0 for def. params
         feed = ExtFeed(feed_type='common',
                        target_cell_type=None,
                        params=p_common[ii],
                        gid=0)
         print(feed)  # test repr
-
-
-def test_external_common_feeds():
-    """Test external common feeds to proximal and distal dendrites."""
-    hnn_core_root = op.join(op.dirname(hnn_core.__file__), '..')
-    params_fname = op.join(hnn_core_root, 'param', 'default.json')
-    params = read_params(params_fname)
-
-    # default parameters have no common inputs (distal or proximal),
-    params.update({'input_dist_A_weight_L2Pyr_ampa': 5.4e-5,
-                   'input_dist_A_weight_L5Pyr_ampa': 5.4e-5,
-                   't0_input_dist': 50,
-                   'input_prox_A_weight_L2Pyr_ampa': 5.4e-5,
-                   'input_prox_A_weight_L5Pyr_ampa': 5.4e-5,
-                   't0_input_prox': 50})
-
-    with Network(deepcopy(params)) as net:
-        net._create_all_spike_sources()
-
-        assert len(net.common_feeds) == 2  # (distal & proximal)
-        for ei in net.common_feeds:
-            assert ei.feed_type == 'common'
-            assert ei.cell_type is None  # artificial cell
-            # parameters should lead to > 0 input spikes
-            assert len(ei.event_times) > 0
-
-            # check that ei.p_ext matches params
-            loc = ei.params['loc'][:4]  # loc=prox or dist
-            for layer in ['L2', 'L5']:
-                key = 'input_{}_A_weight_{}Pyr_ampa'.format(loc, layer)
-                assert ei.params[layer + 'Pyr_ampa'][0] == params[key]
+        assert feed.feed_type == 'common'
+        assert feed.cell_type is None  # artificial cell
+        # parameters should lead to 0 input spikes for default params
+        assert len(feed.event_times) == 0
+        # check that ei.p_ext matches params
+        loc = feed.params['loc'][:4]  # loc=prox or dist
+        for layer in ['L2', 'L5']:
+            key = 'input_{}_A_weight_{}Pyr_ampa'.format(loc, layer)
+            assert feed.params[layer + 'Pyr_ampa'][0] == params[key]
```

Comment:
Since I got rid of the `Network.common_feeds` list attribute, I also had to shuffle around some elements of the common feed test. I ended up moving a few elements to the `test_network.py` module. @cjayb given that you've spend a lot more time thinking about this test, did I do this correctly/adequately?

Reply:
do we lose some information or debuggability because of this change?

Reply:
I like these changes. Took me a while to remind myself of the challenges involved, but I think the new cell class will help in the long run. Ping self at #104

Reply:
I'm also adding some unit tests for the `_ArtificialCell` class (stay tuned for an update). It incorporates most of the same elements I accidentally deleted from the next few lines of code (i.e., lines 65-68). Thanks for looking into this @cjayb!